mbuild/utils/conversion.py

```diff
@@ -1,25 +1,164 @@
+        )
+
+    f0 = 2.0 * (c0 + c1 + c2 + c3 + c4 + c5)
+    if not np.all(np.isclose(f0 / 2, 0, atol=error_tol, rtol=0)):
+        text_for_error_tol = (
+            "f0 = 2 * (c0 + c1 + c2 + c3 + c4 + c5) is not zero. "
+            "The f0/2 term is the constant for the OPLS dihedral. "
+            "Since the f0 term is not zero, the dihedral is not an "
+            "exact conversion; since this constant does not contribute "
+            "to the force equation, this should provide matching results "
+            "for MD, but the energy for each dihedral will be shifted "
+            "by the f0/2 value."
+        )
+        if value_error_out_of_tol is True:
+            raise ValueError("{}".format(text_for_error_tol))
+        elif value_error_out_of_tol is False:
+            warn("WARNING: {}".format(text_for_error_tol))
+
+    f1 = -2 * c1 - (3 * c3) / 2
+    f2 = -c2 - c4
+    f3 = -c3 / 2
+    f4 = -c4 / 4
+    return np.array([f0, f1, f2, f3, f4])
+
+
+def OPLS_to_RB(f0, f1, f2, f3, f4, error_tol=1e-4):
+    r"""Convert OPLS type to Ryckaert-Bellemans type dihedrals.
+
+    .. math::
+    OPLS_torsions &= \frac{f_0}{2} + \frac{f_1}{2}*(1+cos(t)) + \frac{f_2}{2}*(1-cos(2*t)) + \\
+                  &= \frac{f_3}{2}*(1+cos(3*t)) + \frac{f_4}{2}*(1-cos(4*t))
+
+    .. math::
+    RB_{torsions} &= c_0 + c_1*cos(psi) + c_2*cos(psi)^2 + c_3*cos(psi)^3 + \\
+                  &= c4*cos(psi)^4 + c5*cos(5*psi)^5
+
+    where :math:`psi = t - pi = t - 180 degrees`
+
+    Parameters
+    ----------
+    f0, f1, f2, f3, f4 : OPLS dihedrals coeffs (in kcal/mol)
+    error_tol : float, default=1e-4
+        The acceptable absolute tolerance between the OPLS to RB conversion
+        without throwing a warning. Any value entered is converted to an
+        absolute value.
 
     Returns
     -------
-    opls_coeffs : np.array, shape=(4,)
-        Array containing the OPLS dihedrals coeffs f1, f2, f3, and f4
-        (in kcal/mol)
+    RB_coeffs : np.array, shape=(6,)
+        Array containing the Ryckaert-Bellemans dihedrals
+        coeffs c0, c1, c2, c3, c4, and c5 (in kcal/mol)
+
+    Notes
+    -----
+    NOTE: fO IS TYPICALLY NOT IN THE OPLS DIHEDRAL EQUATION (i.e., f0=0).
+
+    .. warning:: The :math:`\frac{f_{0}}{2}` term is the constant for the OPLS dihedral equation,
+    which is and added to a constant for the RB torsions equation via the c0 coefficient.
+    If the f0 term is zero in the OPLS dihedral form or is force set to zero in this equation,
+    the dihedral is may not an exact conversion;
+    since this constant does not contribute to the force equation,
+    this should provide matching results for MD, but the energy for each
+    dihedral will be shifted by the real :math:`\frac{f_{0}}{2}` value.
     """
-    f1 = (-1.5 * c3) - (2 * c1)
-    f2 = c0 + c1 + c3
-    f3 = -0.5 * c3
-    f4 = -0.25 * c4
-    return np.array([f1, f2, f3, f4])
+    try:
+        error_tol = abs(float(error_tol))
+    except:
```

Comment:
again probably want to specify which exception we are trying to catch

Reply:
fixed

Reply:
Refer to my previous comment in the same section but in the other method.

Reply:
Changed